pkg/graph/schema.resolvers.go

```diff
@@ -955,19 +955,38 @@ func (r *mutationResolver) SetRolesForUserOnSystem(ctx context.Context, input mo
 		})
 
 		// wait for both calls to complete
-		err := g.Wait()
-		if err != nil {
+		if err := g.Wait(); err != nil {
 			// don't fail the request if the lookup fails, just log and return from the go func
 			appcontext.ZLogger(emailCtx).Error("failed to lookup user infos for CEDAR notification email", zap.Error(err))
 			return
 		}
 
-		err = r.emailClient.SendCedarRolesChangedEmail(emailCtx, requesterUserInfo.DisplayName, targetUserInfo.DisplayName, rs.DidAdd, rs.DidDelete, rs.RoleTypeNamesBefore, rs.RoleTypeNamesAfter, rs.SystemName, time.Now())
-		if err != nil {
+		if err := r.emailClient.SendCedarRolesChangedEmail(emailCtx, requesterUserInfo.DisplayName, targetUserInfo.DisplayName, rs.DidAdd, rs.DidDelete, rs.RoleTypeNamesBefore, rs.RoleTypeNamesAfter, rs.SystemName, time.Now()); err != nil {
 			// don't fail the request if the email fails, just log and return from the go func
 			appcontext.ZLogger(emailCtx).Error("failed to send CEDAR notification email", zap.Error(err))
 			return
 		}
+
+		if rs.DidAdd {
+			if err := r.emailClient.SendCedarYouHaveBeenAddedEmail(emailCtx, rs.SystemName, input.CedarSystemID, rs.RoleTypeNamesAfter, targetUserInfo.Email); err != nil {
+				// don't fail the request if the email fails, just log and return from the go func
+				appcontext.ZLogger(emailCtx).Error("failed to send CEDAR email to user who was added to team", zap.Error(err))
+				return
+			}
+
+			// send to all except the new team member to avoid duplicate emails
+			// get cedar system info
+			systemDetails, err := r.cedarCoreClient.GetSystemDetail(ctx, input.CedarSystemID)
```

Comment:
Is there a risk in fetching the `systemDetails` _after_ we do the role set? We're passing in `systemDetails.Roles` as "existingTeamMembers", but technically we might already see the newly added team member in that list.

Perhaps we should do 1 or 2 things to address this:
- Make the `r.cedarCoreClient.GetSystemDetail` call _prior_ to the call to `r.cedarCoreClient.SetRolesForUser` at the start of the function
- Make a call to `r.cedarCoreClient.GetRolesBySystem` instead of `r.cedarCoreClient.GetSystemDetail` (since the only thing we actually need from the response is the role list, but `GetSystemDetail` fetches _much_ more than that -- in fact... it might not actually even populate the roles? Perhaps we should sync on this!)

Reply:
thanks again for the call today! updated 😄